.github/workflows/style-and-sp-check.yml

```
@@ -24,8 +24,30 @@ jobs:
         with:
           ref: ${{ github.event.pull_request.head.ref }}
 
-      - name: Install styler
-        run: Rscript -e "install.packages('styler')"
+      - name: Install packages
+        run: Rscript -e "install.packages(c('styler', 'spelling'))"
+
+      - name: Run spell check
+        id: spell_check_run
+        run: |
+          Rscript components/spell-check.R
+          results=$(Rscript "components/spell-check.R")
+          echo "::set-output name=sp_chk_results::$results"
+
+      - name: Archive spelling errors
+        uses: actions/upload-artifact@v2
+        with:
+          name: spell-check-results
+          path: components/spell_check_results.tsv
+
+      # If there are too many spell checks, this will stop everything
+      - name: Check on spell check results
+        if: ${{ steps.spell_check_run.outputs.sp_chk_results > 340 }}
+        run: |
+          echo Too many spelling errors, $result
+          exit 1
+        env:
+          result: ${{ steps.spell_check_run.outputs.sp_chk_results }}
```

I think we should print out the spelling errors here, since it will be easier to see them here than by downloading a zip file, etc. Which means I think we should make this run every time, and use the shell script conditional to kill the workflow.

Also, I moved the `env:` declaration up for my preferences (I want to see variables declared before use).

```suggestion
      - name: Check on spell check results
        env:
          result: ${{ steps.spell_check_run.outputs.sp_chk_results }}
        run: |
          cat components/spell_check_results.tsv
          if (( $result > 340 )); then
            echo Too many spelling errors, $result
            exit 1
          fi
```